pilot/model/validation.go

```diff
@@ -873,6 +876,12 @@ func ValidateEgressRule(msg proto.Message) error {
 		if err := ValidateEgressRulePort(port); err != nil {
 			errs = multierror.Append(errs, err)
 		}
+
+		if cidrDestinationService && Protocol(strings.ToUpper(port.Protocol)) != ProtocolTCP {
```

Comment:
you can do CIDR for mongo/tcp as they use the same filter. You cannot use CIDR for redis. You can use host/IP. Its just like http

Reply:
And need to allow https as well

Reply:
this PR adds support for TCP only. Note that any protocol on top of TCP can use it by specifying tcp in the egress rule. This way you can specify tcp for an https connection, on port 443, and the envoy will serve as a TCP proxy. In the current implementation, https means TLS origination by the envoy, and http from a microservice to the envoy.

I will add support for Redis and Mongo filter in subsequent PRs.